sound/soc/sof/loader.c

```diff
@@ -496,7 +575,20 @@ int snd_sof_load_firmware_raw(struct snd_sof_dev *sdev)
 			fw_filename);
 	}
 
-	plat_data->fw_offset = 0;
+	/* check for extended manifest */
+	ext_man_size = snd_sof_ext_man_size(plat_data->fw->data);
+	if (ext_man_size > 0) {
```

Comment:
if you modify the function to check for invalid header, it will also be able to return negative error codes. Then you'll have to process `>0`, `==0` and `<0` cases separately and et `ret = ext_man_size` in the latter case.

Reply:
when function will return negative error codes (after changing return type) this comparison still will be valid. We like to parse `ext_man` only if it is present and valid.